dotCMS/src/main/java/com/dotmarketing/business/RelationshipAPIImpl.java

```diff
@@ -62,13 +62,17 @@ public Relationship byTypeValue(String typeValue) {
 
     @CloseDBIfOpened
     @Override
-    public List<Relationship> byContentType(ContentTypeIf st, boolean hasParent) {
-        return this.relationshipFactory.byContentType(st, hasParent);
+    public List<Relationship> byContentType(ContentTypeIf st, boolean hasParent) throws DotDataException{
+        if(hasParent) {
+            return this.relationshipFactory.byParent(st);
+        }else{
+            return this.relationshipFactory.byChild(st);
+        }
```

Comment:
I think we should not throw exception from `relationshipFactory.byParent` or `this.relationshipFactory.byChild`, since the new methods at factory level catch the exception and log, so we should be consistent here. Also to maintain the original method signature.

Reply:
At factory level, those methods signature (byParent and byChild) has not been changed https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/contenttype/business/RelationshipFactory.java#L19-L21

The method previously called was deleted, so we need to call these 2 instead directly.